internal/cmd/controller/controllers/cluster/import.go

```diff
@@ -85,27 +90,48 @@ func (i *importHandler) onConfig(config *config.Config) error {
 		return nil
 	}
 
-	for _, cluster := range clusters.Items {
-		if cluster.Spec.KubeConfigSecret == "" {
-			continue
-		}
-		secret, err := i.secrets.Get(cluster.Namespace, cluster.Spec.KubeConfigSecret)
-		if err != nil {
-			return err
-		}
-		if string(secret.Data["apiServerURL"]) == "" || string(secret.Data["apiServerCA"]) == "" {
-			logrus.Debugf("API server fallback-config changed, trigger cluster import for cluster %s/%s", cluster.Namespace, cluster.Name)
-			c := cluster.DeepCopy()
-			c.Status.AgentConfigChanged = true
-			_, err := i.clusters.UpdateStatus(c)
-			if err != nil {
-				return err
+	b := &backoff.Backoff{
+		Min:    1 * time.Second,
+		Max:    3 * time.Minute,
+		Factor: 2,
+		Jitter: true,
+	}
+
+	go func() {
```

Comment:
why do we use a goroutine here? I think this might trigger more simultaneous calls